fineract-provider/src/main/java/org/apache/fineract/cob/service/InlineLoanCOBExecutorServiceImpl.java

```diff
@@ -120,6 +132,19 @@ public void execute(List<Long> loanIds, String jobName) {
         }
     }
 
+    private LocalDate getOldestCOBBusinessDate(List<Long> loanIds) {
+        return loanRepository.findOldestCOBProcessedLoanByLoanIds(loanIds)
+                .orElse(ThreadLocalContextUtil.getBusinessDateByType(BusinessDateType.COB_DATE).minusDays(1));
+    }
+
+    private List<Long> getLoanIdsTOBeProcessed(List<Long> loanIds, LocalDate oldestCOBBusinessDate, LocalDate cobBusinessDate) {
+        if (!oldestCOBBusinessDate.isBefore(cobBusinessDate)) {
+            return Collections.emptyList();
+        } else {
+            return loanRepository.findAllNonClosedLoanIdsOneDayBehindByLoanIds(cobBusinessDate, loanIds);
```

Comment:
Is this correct? Why do we fetch only the loans that one day behind? We should fetch all of them, no?